src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,199 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static final int FALSE_INDEX = -1;
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        areIndexInSizeRange(index);
+        linkReassignmentForAdd(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final T removedItem = node(index).item;
+        node(index).item = value;
+        return removedItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        final T removedItem = node(index).item;
+        linkReassignmentForRemove(index);
+        return removedItem;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findIndex(object);
+        if (index == FALSE_INDEX) {
+            return false;
+        }
+        linkReassignmentForRemove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void areIndexInSizeRange(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " doesn't exist");
+        }
+    }
+
+    private void areIndexInRange(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + " doesn't exist");
+        }
+    }
+
+    private int findIndex(T object) {
```

Comment:
better implement this method in this way
```suggestion
    private Node<T> findByIndex(T object) {
```
the info about index is not needed.